src/main/kotlin/net/maizegenetics/phgv2/cli/ExportVcf.kt

```diff
@@ -1,15 +1,51 @@
 package net.maizegenetics.phgv2.cli
 
 import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.groups.mutuallyExclusiveOptions
+import com.github.ajalt.clikt.parameters.options.convert
 import com.github.ajalt.clikt.parameters.options.default
 import com.github.ajalt.clikt.parameters.options.option
 import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.groups.required
+import com.github.ajalt.clikt.parameters.groups.single
 import net.maizegenetics.phgv2.utils.verifyURI
 import org.apache.logging.log4j.LogManager
 import java.io.File
 
+
+
+// Enum indicating type of input for sample names
+enum class SampleFormatEnum(private val value: String) {
+    FILE("File"), LIST("List");
+}
+
 /**
- * This will export the give samples dataset to a hvcf file.
+ * Sealed classes to handle either a list of sample names or a file containing sample names, 1 per line.
+ */
+sealed class SampleProcessing {
+   // abstract fun getSampleOption(): Pair<SampleFormatEnum,String>
+    // Would be nice if this could return the command itself, but
+    // the command requires the value of the other input parameters,
+    // which are not known here.
+    abstract fun getExportCommand(): List<String>
+    data class SampleFile(val sampleFile: String): SampleProcessing() {
+        @Override
+        override fun getExportCommand(): List<String> {
```

Comment:
I think this would be better if it returned Pair<SampleFormatEnum.FILE, List<String>>. That way you don't need to index into the list, you would just need to join the string array.

Reply:
I had issues with Pair<> - got errors relating to constructor needing to be initialized ( or something like that, I forget the actual message).  It would not take Pair in this setting, so I switched to List.